numba/tests/test_array_methods.py

```diff
@@ -917,6 +921,18 @@ def check(a):
         check(np.array([[3.1, 3.1], [1.7, 2.29], [3.3, 1.7]]))
         check(np.array([]))
 
+    def test_repeat(self):
+        pyfunc = np_repeat
+        cfunc = jit(nopython=True)(pyfunc)
+
+        def check(a, repeats):
+            np.testing.assert_equal(pyfunc(a, repeats), cfunc(a, repeats))
+
+        check(np.array([3]), repeats=4)
```

Comment:
These tests are a good start but need expanding to cover:
 * repeats = 0
 * repeats = negative number
 * repeats = non-integer
 * a = some high dimensioned array
 * a = empty 2d array
 * a = complex data type array
 * a = not a supported type

The use of `itertools.product` may be beneficial for trying out the combinations.

Reply:
Some of these testcases are not supported by repeat in numpy too. But the errors coming out are different in my code and theirs.